examples/flava/native/train.py

```diff
@@ -156,33 +168,50 @@ def create_model(self) -> torch.nn.Module:
             mp = None
             if self.config.training.enable_half_reduce_in_fsdp:
                 mp = MixedPrecision(
-                    # param_dtype=self.half_dtype,  not working
+                    #param_dtype=self.half_dtype,  #not working
                     reduce_dtype=self.half_dtype,
-                    # buffer_dtype=self.half_dtype,
+                    #buffer_dtype=self.half_dtype,
                 )
 
-            model = FSDP(
-                model,
-                mixed_precision=mp,
-                auto_wrap_policy=partial(
-                    transformer_auto_wrap_policy,
-                    transformer_layer_cls={TransformerEncoderLayer},
-                ),
-            )
+            wrapper_cls = {
+                    TransformerEncoderLayer,
+                    ImageTransformer,
+                    FLAVATransformerWithoutEmbeddings,
+                    DalleVAEEncoder,
+                    DalleEncoderBlock,
+                    BERTTextEncoder,
+                    TransformerEncoder,
+            #        BERTTextEmbeddings,
+                    }
+            wrapper_cls = {
+                    TransformerEncoderLayer,
+                    ImageTransformer,
+                    BERTTextEncoder,
+                    DalleVAEEncoder,
+                    DalleEncoderBlock,
+            }
+            #wrapper_cls = {TransformerEncoderLayer}
+            print(f"mp config {mp}")
 
             if self.config.training.activation_checkpointing:
                 # note: activation checkpointing wrapper currently is faulty
```

Comment:
N00b Q: since you put these comments here, would you mind explaining:
- As FSDP by itself distributes a large model across GPUs, is it still necessary to use activation checkpointing?
- When would you want to use `re-entrant` vs. `non re-entrant`?
- Any idea why memory reduction from checkpointing is less than expected?

Thanks!